### PR TITLE
Laboratory work #1, Nikita Gruzdev - 23FPL2

### PR DESCRIPTION
**Pull request by KolkiRik**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -20,6 +23,14 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    text = text.lower()
+    split1 = text.split()
+    split2 = []
+    for word in split1:
+        for token in word:
+            if token.isalpha():
```
[the remark this answers is not in the document]
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py]: fixed?

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -20,6 +23,14 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    text = text.lower()
+    split1 = text.split()
```
[the remark this answers is not in the document]
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -20,6 +23,14 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    text = text.lower()
```
[the remark this answers is not in the document]
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/start.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 Language detection starter
 """
 # pylint:disable=too-many-locals, unused-argument, unused-variable
-
+from main import tokenize;
```
[the remark this answers is not in the document]
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/start.py]: removed the semicolon, but the import returns an error: 
No module named 'lab_1_classify_by_unigrams'

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -20,6 +23,18 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if type(text) == str:
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:26]: use isinstance to check type

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -20,6 +23,18 @@ def tokenize(text: str) -> list[str] | None:
...
+        text = text.lower().split()
+        split2 = []
+        for word in text:
+            for token in word:
+                if not token.isalpha():
+                    continue
+                else:
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:33]: else is not required, enough write split2 += token

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -20,6 +23,18 @@ def tokenize(text: str) -> list[str] | None:
...
+                    split2 += token
+        return (split2)
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:35]: remove the parentheses

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (Vasilisa-Blyudova) [reply lab_1_classify_by_unigrams/main.py:26]: return none

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -34,6 +49,23 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if type(tokens) != list:
+        return None
+    else:
+        for token in tokens:
+            if type(token) != str:
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:56]: the same

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -34,6 +49,23 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if type(tokens) != list:
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:52]: the same

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -34,6 +49,23 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        freq = {}
+        prop = {}
+        total_tokens = len(tokens)
+        for token in tokens:
+            if token in freq:
+                freq[token] += 1
+            else:
+                freq[token] = 1
+        for key, value in freq.items():
+            prop[key] = value / total_tokens
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:67]: use compound operator /=

File: lab_1_classify_by_unigrams/main.py [hunk 11]
```diff
@@ -49,6 +81,13 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
 
     In case of corrupt input arguments, None is returned
     """
+    if type(language) != str or type(text) != str:
+        return None
+    else:
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:86]: else is not required

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -64,6 +103,13 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not isinstance(predicted, list) or not isinstance(actual,list) or (len(predicted) != len(actual)):
+        return None
+    else:
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:108]: else is not required

File: lab_1_classify_by_unigrams/main.py [hunk 13]
```diff
@@ -64,6 +103,13 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
+        sum_diff = 0
+        for i, n in enumerate(actual):
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:110]: rename variables

File: lab_1_classify_by_unigrams/main.py [hunk 14]
```diff
@@ -84,6 +130,24 @@ def compare_profiles(
     In case of corrupt input arguments or lack of keys 'name' and
     'freq' in arguments, None is returned
     """
+    if not (isinstance(unknown_profile, dict)) or not (isinstance(profile_to_compare, dict)):
+        return None
+    else:
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:135]: the same

File: lab_1_classify_by_unigrams/main.py [hunk 15]
```diff
@@ -84,6 +130,24 @@ def compare_profiles(
...
+                unknown_profile['freq'][i] = 0
+        sorted_unknown = dict(sorted(unknown_profile['freq'].items()))
+        sorted_compare = dict(sorted(profile_to_compare['freq'].items()))
+        sort_comp_list = []
+        sort_unkn_list = []
+        for i in sorted_unknown.values():
+            sort_unkn_list.append(i)
+        for i in sorted_compare.values():
+            sort_comp_list.append(i)
+        return round(calculate_mse(sort_unkn_list, sort_comp_list),3)
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:150]: no need to round values

**Review by Vasilisa-Blyudova — COMMENTED** [review]
Good job! Raise score to 10!

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Vasilisa-Blyudova) [reply lab_1_classify_by_unigrams/main.py]: yes

File: lab_1_classify_by_unigrams/start.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vasilisa-Blyudova) [reply lab_1_classify_by_unigrams/start.py]: try to write from lab_1_classify_by_unigrams.main import tokenize

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:26]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:33]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:35]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:56]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 9 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:52]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 11 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:86]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 12 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:108]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 13 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:110]: renamed

File: lab_1_classify_by_unigrams/main.py (hunk 14 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:135]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 15 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:150]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 16]
```diff
@@ -1,6 +1,9 @@
 """
 Lab 1.
 
+
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:4]: rm

File: lab_1_classify_by_unigrams/main.py [hunk 17]
```diff
@@ -20,6 +23,15 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not isinstance(text, str):
+        return None
+    split_text = text.lower().split()
+    split2 = []
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:29]: rename

File: lab_1_classify_by_unigrams/main.py [hunk 18]
```diff
@@ -34,6 +46,22 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
     In case of corrupt input arguments, None is returned
     """
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
+        if not isinstance(token, str):
+            return None
+    freq = {}
+    prop = {}
+    total_tokens = len(tokens)
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:56]: don't create a new variable if you only use it once

**Review by Vasilisa-Blyudova — COMMENTED** [review]
Great work! Next goal is a score of 10!

File: lab_1_classify_by_unigrams/main.py (hunk 16 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py:4]: rmed

File: lab_1_classify_by_unigrams/main.py (hunk 17 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py]: renamed

File: lab_1_classify_by_unigrams/main.py (hunk 18 above)
Comment (KolkiRik) [reply lab_1_classify_by_unigrams/main.py]: removed

**Comment by KolkiRik** [comment]
Confused by the last error, AssertionError in detect_language_advanced, unsure what's causing it

**Pull request closed without merging**